extension/storage/nop_client.go

```diff
@@ -0,0 +1,29 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+type NopClient struct{}
```

Comment:
Please follow the model we have in core. and provide a `func NewNopClient() Client` and avoid exposing the new struct.

May also want to have this in a `storagetest` package, can be done later.

Reply:
I've updated this to follow the pattern established in core. However, I have a question about potentially keeping `NewNopClient` available.

One of the expectations is that components that make use of a storage extension should seamlessly handle the situation where no storage extension is configured. My intention with introducing this `nopClient` was to provide an easy way for components to fulfill this expectation. Consider:

```go
// Start the component
func (c *myComponent) Start(ctx context.Context, host component.Host) error {
    client := getStorageClient(ctx, host)
    
    // component implementation uses client but doesn't care if it's nop or not
}


func getStorageClient(ctx context.Context, host component.Host) storage.Client {
    for _, ext := range host.GetExtensions() {
        if ext is a storage extension {
            return ext.GetClient()
        }
    }
    return storage.NewNopClient()
}
```

Do you think this is a useful pattern? I suppose components can implement their own `nopClient` if necessary, but I thought it might be helpful to include.

Reply:
Let's see in practice how this will be used in few examples, probably we should make sure we re-evaluate this after couple of PRs that uses it.